File: framework/web-core/src/main/java/top/bettercode/simpleframework/servlet/HandlerMethodContextHolderConfiguration.java
package top.bettercode.simpleframework.servlet;

import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerMapping;

@Configuration(proxyBeanMethods = false)
@ConditionalOnWebApplication
public class HandlerMethodContextHolderConfiguration {


  public HandlerMethodContextHolderConfiguration(
      RequestMappingHandlerMapping requestMappingHandlerMapping) {
    HandlerMethodContextHolder.setHandlerMapping(requestMappingHandlerMapping);
  }


}

File: web/src/main/kotlin/top/bettercode/summer/web/servlet/HandlerMethodContextHolderConfiguration.kt
package top.bettercode.summer.web.servlet

import org.springframework.boot.autoconfigure.condition.ConditionalOnBean
import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication
import org.springframework.context.annotation.Configuration
import org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerMapping

@ConditionalOnBean(RequestMappingHandlerMapping::class)
@Configuration(proxyBeanMethods = false)
@ConditionalOnWebApplication
class HandlerMethodContextHolderConfiguration(
        requestMappingHandlerMapping: RequestMappingHandlerMapping) {
    init {
        HandlerMethodContextHolder.setHandlerMapping(requestMappingHandlerMapping)
    }
}